clang/test/SemaSYCL/union-kernel-param.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -ast-dump %s | FileCheck %s
+// expected-no-diagnostics
+
+// This test checks that compiler generates correct kernel arguments for
+// union without array.
+
+#include <sycl.hpp>
+
+using namespace cl::sycl;
+
+typedef float realw;
+
+typedef union dpct_type_54e08f {
```

Comment:
Can you add more than one field in this union? Also why do we need the typedefs? A simple test with 2-3 non-array fields will be best to check basic functionality.

Reply:
Yep!  I also want to see an:

array of unions. (should be 1 union per element)

Array of structs-with-unions. (should be decomposed structs, but unions still only a single).

Reply:
I will update tests.